internal/pkg/task/env_runner.go

```diff
@@ -53,16 +54,16 @@ func (r *EnvRunner) Run() ([]*Task, error) {
 		return nil, fmt.Errorf("get cluster for environment %s: %w", r.Env, err)
 	}
 
-	filters := r.filtersForVPCFromAppEnv()
-
-	subnets, err := r.VPCGetter.PublicSubnetIDs(filters...)
+	description, err := r.EnvironmentDescriber.Describe()
 	if err != nil {
-		return nil, fmt.Errorf(fmtErrPublicSubnetsFromEnv, r.Env, err)
+		return nil, fmt.Errorf(fmtErrDescribeEnvironment, r.Env, err)
 	}
-	if len(subnets) == 0 {
+	if description == nil || description.EnvironmentVPC == nil || len(description.EnvironmentVPC.PublicSubnetIDs) == 0 {
 		return nil, errNoSubnetFound
```

Comment:
I think we don't need to check `description`.
Because when `description` is nil, we'll get error and return above line.

And I don't use pointer for `description.EnvironmentVPC` .
Because when we can't get data from Stack, [we return error](https://github.com/aws/copilot-cli/pull/2024/commits/1664aa98a34280afd51089710b631d7f2859f21b#diff-7e9e5614d4159f2672b5984f480c4b8aaa5e2c139ed2de7f948d283103de6fb5L137) so we can set this struct always.

Reply:
Got it! Thanks for the follow up!